flink-yarn/src/main/java/org/apache/flink/yarn/entrypoint/YarnDeploymentTargetNames.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+	package org.apache.flink.yarn.entrypoint;
+
+import org.apache.flink.annotation.Internal;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * A class containing all the supported deployment target names for Yarn.
+ */
+@Internal
+public enum YarnDeploymentTargetNames {
```

Comment:
I think we could move this `enum` to `YarnConfigOptions`, rather than in the package `org.apache.flink.yarn.entrypoint`. It will be more appropriate. It just my personal preference. You could keep it.

Reply:
Moved it to its own class.